pandas/tests/series/methods/test_isin.py

```diff
@@ -234,3 +234,17 @@ def test_isin_filtering_with_mixed_object_types(data, is_in):
     expected = Series([True, False])
 
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [([1, 2, 3]), ([1.0, 2.0, 3.0])],
+)
+def test_isin_filtering_on_iterable(data):
+    # GH 50234
+
+    ser = Series(data)
+    result = ser.isin(i for i in [1, 2])
```

Comment:
sorry, I didn't catch this the first time round - doesn't the original issue have `(i for i in [1.0, 2.0])`?

Reply:
Agreed, we should parametrize over both, e.g. [1, 2] and [1.0, 2.0]

Reply:
Good point! Updated parameters accordingly.